### PR TITLE
KAFKA-6302: Improve AdmintClient JavaDocs

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @cmccabe

File: clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public DeleteTopicsResult deleteTopics(Collection<String> topics) {
      *
      * This operation is not transactional so it may succeed for some topics while fail for others.
      *
-     * It may take several seconds after AdminClient#deleteTopics returns
+     * It may take several seconds after the {@code DeleteTopicsResult} returns
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java:141]: Isn't the same issue present in the other topic management methods?

Comment (mjsax) [reply clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java:141]: Fair point. But the others don't mention anything like this (except `createTopics`). Thus updating `createTopics` seems to make sense (to at least avoid miss leading phrases).

I would leave all others as-is.

**Comment by mjsax** [comment]
Updated this.

File: clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java (hunk 1 above)
Comment (cmccabe) [reply clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java:141]: I agree that the same issue is present in many other admin methods.  However, `createTopics` and `deleteTopics` tend to be the ones that surprise users.  Users do things like create a topic and then expect to be able to delete it immediately, in a test.

**Comment by cmccabe** [comment]
@mjsax: LGTM

**Comment by mjsax** [comment]
Retest this please.

Can this be merged than @ijuma ?

**Review by ijuma — APPROVED** [review]
LGTM, will merge after Jenkins completes.

**Comment by mjsax** [comment]
Retest this please.

**Pull request merged**